README.md

```diff
@@ -1,161 +1,87 @@
 
-### Secondary goals
+```js
+(function () {
+  function foo () {
+    console.log( 'this function was included!' );
+  }
+
+  function bar () {
+    console.log( 'this function was not' );
+    baz();
+  }
+
+  function baz () {
+    console.log( 'neither was this' );
+  }
+
+  foo();
+})();
+```
 
-* Support for legacy module formats
-* Respect for original formatting and code comments
+A minifier can detect that `foo` gets called, but that `bar` doesn't. When we remove `bar`, it turns out that we can also remove `baz`.
 
+Unfortunately, **traditional modules – CommonJS and AMD – make this kind of optimisation next to impossible**. Rather than *excluding dead code*, we should be *including live code*. That's only possible with ES6 modules.
 
-### API
 
-The example below is aspirational. It isn't yet implemented - it exists in the name of [README driven development](http://tom.preston-werner.com/2010/08/23/readme-driven-development.html).
+## What's the catch?
 
-```js
-rollup.rollup({
-  // The bundle's starting point
-  entry: 'app.js',
-
-  // Any external modules you don't want to include
-  // in the bundle (includes node built-ins)
-  external: [ 'path', 'fs', 'some-other-lib' ]
-}).then( function ( bundle ) {
-  // generate code and a sourcemap
-  const { code, map } = bundle.generate({
-    // output format - 'amd', 'cjs', 'es6', 'iife', 'umd'
-    format: 'amd',
-
-    // exports - 'auto', 'none', 'default', 'named'
-    exports: 'auto',
-
-    // amd/umd options
-    moduleId: 'my-library',
-
-    // umd options
-    moduleName: 'MyLibrary', // necessary if the bundle has exports
-    globals: {
-      backbone: 'Backbone'
-    }
-  });
-
-  fs.writeFileSync( 'bundle.js', code + '\n//# sourceMappingURL=bundle.js.map' );
-  fs.writeFileSync( 'bundle.js.map', map.toString() );
-
-  // possible convenience method
-  bundle.write({
-    dest: 'bundle.js', // also writes sourcemap
-    format: 'amd'
-  });
-});
-```
+Most libraries that you depend on aren't written as ES6 modules, so Rollup can't work with them directly. (You *can* bundle your own app or library code with Rollup as a CommonJS module, then pass the result over to Webpack or Browserify, of course.)
 
-The duplication (`rollup.rollup`) is intentional. You have to say it like you're in the circus, otherwise it won't work.
+**You can help!** It's possible to write libraries as ES6 modules while still making it easy for other developers to use your code as they already do, using the [jsnext:main](https://github.com/rollup/rollup/wiki/jsnext:main) field in your package.json. You'll be writing your code in a more future-proof way, and helping to bring an end to the [dark days of JavaScript package management](https://medium.com/@trek/last-week-i-had-a-small-meltdown-on-twitter-about-npms-future-plans-around-front-end-packaging-b424dd8d367a).
```

Comment:
This call to action is good, but I think we need an example project that shows how this should be done. I created one that you can link to if you like: https://github.com/eventualbuddha/rollup-starter-project. If you have suggestions on how to improve that starter project (i.e. use `rollup-babel`?) please open an issue or PR!